src/librustc_mir/transform/acs_propagate.rs

```diff
@@ -40,25 +42,49 @@ use pretty;
 
 #[derive(PartialEq, Debug, Eq, Clone)]
 enum Either<'tcx> {
-    Top,
     Lvalue(Lvalue<'tcx>),
     Const(Constant<'tcx>),
 }
 
-impl<'tcx> Lattice for Either<'tcx> {
-    fn bottom() -> Self { unimplemented!() }
+#[derive(Debug, Clone)]
+enum AcsLattice<'tcx> {
+    Bottom,
```

Comment:
The bottom element in this lattice is an empty HashMap. It is fine to have it this way, because `HashMap::new()` does not allocate, to my knowledge.

Similarly, I’m confused as to why the `Top` was removed from `Either`. I do not see how you could merge two different constants together and produce anything other than a `Top`.

/me shrugs